src/librustc_mir/transform/fragment_locals.rs

```diff
@@ -0,0 +1,397 @@
+use crate::transform::{MirPass, MirSource};
+use rustc::mir::visit::{MutVisitor, NonUseContext, PlaceContext, Visitor};
+use rustc::mir::*;
+use rustc::ty::layout::VariantIdx;
+use rustc::ty::util::IntTypeExt;
+use rustc::ty::{self, Ty, TyCtxt};
+use rustc_index::vec::IndexVec;
+use rustc_span::Span;
+use std::collections::BTreeMap;
+use std::iter::Step;
+use std::ops::Range;
+
+pub struct FragmentLocals;
```

Comment:
Can we get a doc comment added either to the MIR pass or to this module explaining what this does? Imo, the PR body does a great job explaining this and could probably be adapted as a doc comment.

Reply:
Note @nikomatsakis' original review comments - I still need to go through them.